internal/installation/paths.go

```diff
@@ -29,23 +29,38 @@ func InstallPath() (string, error) {
 	if !condition.BuiltViaCI() && strings.Contains(os.Args[0], "/build/") {
 		return filepath.Dir(os.Args[0]), nil
 	}
-	if path, ok := os.LookupEnv(constants.OverwriteDefaultInstallationPathEnvVarName); ok {
-		return path, nil
-	}
 
 	// If State Tool is already exists then we should detect the install path from there
 	stateInfo := appinfo.StateApp()
-	activeStateOwnedPath := strings.Contains(strings.ToLower(stateInfo.Exec()), "activestate")
-	if fileutils.TargetExists(stateInfo.Exec()) {
-		if filepath.Base(filepath.Dir(stateInfo.Exec())) == BinDirName && activeStateOwnedPath {
-			return filepath.Dir(filepath.Dir(stateInfo.Exec())), nil // <return this>/bin/state.exe
-		}
+	if !fileutils.TargetExists(stateInfo.Exec()) {
 		return filepath.Dir(stateInfo.Exec()), nil // <return this>/state.exe
 	}
 
+	activeStateOwnedPath := strings.Contains(strings.ToLower(stateInfo.Exec()), "activestate")
+	if filepath.Base(filepath.Dir(stateInfo.Exec())) == BinDirName && activeStateOwnedPath {
+		return filepath.Dir(filepath.Dir(filepath.Dir(stateInfo.Exec()))), nil // <return this>/<branch>/bin/state.exe
+	}
+
 	return DefaultInstallPath()
 }
 
+func BranchPathFromInstallPath(branch string) (string, error) {
```

Comment:
This seems inverted; you're returning the path for a branch, not a branch for a path. AAlso this feels like it overlaps with `installation.InstallPathForBranch`